app/code/Magento/InventoryCatalog/Ui/DataProvider/Product/Listing/Modifier/QuantityPerSource.php

```diff
@@ -43,7 +43,9 @@ public function modifyData(array $data)
     public function modifyMeta(array $meta)
     {
         if (true === $this->isSingleSourceMode->execute()) {
-            $meta['product_columns']['children']['quantity_per_source']['arguments'] = null;
+            if (isset($meta['product_columns']['children']['quantity_per_source']['arguments'])) {
```

Comment:
Looks like it is the wrong fix
Because `$meta` is always empty at this place

You can recheck scenario from
https://github.com/magento-engcom/msi/issues/607

Reply:
The same modifier is called from several grids. In some cases there are `children` with the `quantity_per_source` and/or `salable_quantity` key. In this cases we need to set `arguments` to `null` if single source mode is used. In other cases this keys are missing and if we add them in any case this leads in a Exception while rendering the UI components... 

I see two ways to fix it:
1. We need to add the fields `quantity_per_source` and `salable_quantity` to any grid which use this `Magento\InventoryCatalog\Ui\DataProvider` 
2. We manipulate this fields only if they are already present

Do I miss understand something?

Reply:
Also we decided no to delete `salable_quantity` column for single source mode. So, @vadimjustus, you can delete class `Magento/InventoryCatalog/Ui/DataProvider/Product/Listing/Modifier/SalableQuantity` and its mention on `di.xml`.